src/hooks/use-fetch-wpcom-sites/index.tsx

```diff
@@ -82,25 +92,29 @@ export function transformSingleSiteResponse(
 	};
 }
 
-function transformSiteResponse(
-	sites: SitesEndpointSite[],
-	connectedSiteIds: number[]
-): SyncSite[] {
-	return sites.reduce( ( acc: SyncSite[], site ) => {
-		if ( site.is_deleted && ! connectedSiteIds.some( ( id ) => id === site.ID ) ) {
-			return acc;
-		}
+export function transformSiteResponse( sites: unknown[], connectedSiteIds: number[] ): SyncSite[] {
+	return sites
+		.map( ( rawSite ) => {
+			try {
+				const site = sitesEndpointSiteSchema.parse( rawSite );
 
-		acc.push( transformSingleSiteResponse( site, getSyncSupport( site, connectedSiteIds ) ) );
+				if ( site.is_deleted && ! connectedSiteIds.some( ( id ) => id === site.ID ) ) {
+					return null;
+				}
 
-		return acc;
-	}, [] );
+				return transformSingleSiteResponse( site, getSyncSupport( site, connectedSiteIds ) );
+			} catch ( error ) {
+				Sentry.captureException( error );
+				return null;
+			}
+		} )
+		.filter( ( site ): site is SyncSite => site !== null );
```

Comment:
`site is SyncSite` - personally for me looks not very nice and looks more wordy, having extra step with filtering. I dont' have strong preference, but take a look one more time, maybe `reduce` would be more simple way here

Reply:
I typically find chained `filter` and `map` calls easier to read, but I also don't have a strong preference. I had to make additional tweaks to accommodate my changes from https://github.com/Automattic/studio/pull/897, but I've reverted the first call to a `.reduce`